ansi/truncate.go

```diff
@@ -12,8 +12,13 @@ import (
 // This function is aware of ANSI escape codes and will not break them, and
 // accounts for wide-characters (such as East Asians and emojis).
 func Truncate(s string, length int, tail string) string {
-	tw := StringWidth(tail)
-	length -= tw
+	if sw := StringWidth(s); sw == length {
```

Comment:
Unsure to understand why you need to check the exact length.

Then you go through all the code.

I would have expected something like this

```diff
-if sw := StringWidth(s); sw == length
+if sw := StringWidth(s); sw <= length {
+       return s
+}
-        tail = "" 
-} else { 
-          tw := StringWidth(tail) 
-          length -= tw 
-}
+tw := StringWidth(tail) 
+length -= tw 
```

Reply:
Ah, good catch! I'll update the commit and benchmarks

Reply:
You will have yo update the commit message and add unit test for this.


something like

"foo", 5, "..."

Reply:
added test case